mongo/options/createcollectionoptions.go

```diff
@@ -0,0 +1,226 @@
+// Copyright (C) MongoDB, Inc. 2017-present.
+//
+// Licensed under the Apache License, Version 2.0 (the "License"); you may
+// not use this file except in compliance with the License. You may obtain
+// a copy of the License at http://www.apache.org/licenses/LICENSE-2.0
+
+package options
+
+// DefaultIndexOptions represents the default options for a collection to apply on new indexes. This type can be used
+// when creating a new collection through the CreateCollectionOptions.SetDefaultIndexOptions method.
+type DefaultIndexOptions struct {
+	// Specifies the storage engine to use for the index. The value must be a document in the form
+	// {<storage engine name>: <options>}. The default value is nil, which means that the default storage engine
+	// will be used.
+	StorageEngine interface{}
+}
+
+// DefaultIndex creates a new DefaultIndexOptions instance.
+func DefaultIndex() *DefaultIndexOptions {
+	return &DefaultIndexOptions{}
+}
+
+// SetStorageEngine sets the value for the StorageEngine field.
+func (d *DefaultIndexOptions) SetStorageEngine(storageEngine interface{}) *DefaultIndexOptions {
+	d.StorageEngine = storageEngine
+	return d
+}
+
+// CreateCollectionOptions represents options that can be used to configure a CreateCollection operation.
+type CreateCollectionOptions struct {
+	// Specifies whether or not an index should automatically be created on the _id field. This option has been
+	// deprecated in MongoDB version 3.2 and removed in MongoDB version 4.0. The default value is true.
+	//
+	// TODO: in 4.0, the option was not fully removed, but can only be used when creating a collection in the "local"
+	// database. Should this be mentioned in the docs above?
+	AutoIndexID *bool
```

Comment:
Just to kick things off.  I don't see a strong reason to include this.  Although it's allowed in older server versions, it is dangerous dangerous (which is why it's been removed) and perhaps more importantly, it's unnecessary for correctness.  And it's possible that it will make upgrades harder, though I'm not sure about that.

Reply:
I'm not sure I agree. It's not advisable to use this option when creating collections, but I don't think it's correct to exclude it from the API either.

Reply:
Since this sounds like something that we really don't want users to actually use, I would lean towards just leaving it out. While I understand wanting full compatibility, there is a precedent with mgocompat for not giving users things that we think are too iffy.

Reply:
Fair enough. Java removed this option in the v4.0 release and Node is planning on doing the same, so I think there's also enough precedent with this specific option to leave it out.